geonode/base/models.py

```diff
@@ -1307,6 +1307,10 @@ def has_thumbnail(self):
     # that indexing (or other listeners) are notified
     def save_thumbnail(self, filename, image):
         upload_path = os.path.join('thumbs/', filename)
+        local_thumbs = os.path.join(settings.MEDIA_ROOT, "thumbs")
+        if not os.path.exists(local_thumbs):
+            os.makedirs(local_thumbs)
+
```

Comment:
This needs fixing as discussed. If we want to be storage independent, all file and path changes should go through Django's storage system.